docs/source/user_guide/best_practices.rst

```diff
@@ -170,3 +170,72 @@ this operation should be avoided.
 
 See `Operations on different DataFrames <options.rst#operations-on-different-dataframes>`_ for more details.
 
+
+Use Koalas APIs directly whenever possible
+------------------------------------------
+
+Although Koalas has most of the pandas-equivalent APIs with pandas, there are several APIs not implemented yet or explicitly unsupported.
+
+As an example, Koalas does not implement ``__iter__()`` to prevent users to collect all data into the client (driver) side from the whole cluster.
+Unfortunately, many external APIs such as Python built-in functions such as min, max, sum, etc. require the given argument to be iterable.
+In case of pandas, it works properly out of the box as below:
+
+.. code-block:: python
+
+   >>> import pandas as pd
+   >>> max(pd.Series([1, 2, 3]))
+   3
+   >>> min(pd.Series([1, 2, 3]))
+   1
+   >>> sum(pd.Series([1, 2, 3]))
+   6
+
+pandas dataset lives in the single machine, and is naturally iterable locally within the same machine.
+However, Koalas dataset lives across multiple machines, and they are computed in a distributed manner.
+It is difficult to be locally iterable and it is very likely users collect the entire data into the client side without knowing it.
+Therefore, it is best to stick to using Koalas APIs.
+The examples above can be converted as below:
+
+.. code-block:: python
+
+   >>> import databricks.koalas as ks
+   >>> ks.Series([1, 2, 3]).max()
+   3
+   >>> ks.Series([1, 2, 3]).min()
+   1
+   >>> ks.Series([1, 2, 3]).sum()
+   6
+
+Another common pattern from pandas users might be to rely on list comprehension or generator expression.
+However, it also assumes the dataset is locally iterable under the hood.
+Therefore, it works seamlessly in pandas as below:
+
+.. code-block:: python
+
+   >>> import pandas as pd
+   >>> data = []
+   >>> pser = pd.Series([20, 21, 12],
```

Comment:
I think you can switch `pser = pd.Series([20, 21, 12],` to `pser = pd.Series([20., 21., 12.],`, and then change `kser.apply(lambda x: x ** 2)` in line 236. `apply` uses pandas UDF under the hood, and it requires to launch one job to infer schema. Avoiding these will be much more performant.

Reply:
If you intend to show that `temperature ** 2` can be an arbitrary logic, using `apply` is fine but probably we should replace `temperature ** 2` to something pandas and Koalas series APIs can't do. Also, you might want to specify the type hints for the function given to `apply` to avoid type inference.

Reply:
That's a good point. I learned a lot.

Reply:
I have fixed the example (specify the type hints for the function given to apply to avoid type inference).  also,  I intend to show that `temperature ** 2` can be an arbitrary logic. Is this okay?